src/System Application/App/SFTP Client/src/DotnetSFTPClient.Codeunit.al

```diff
@@ -0,0 +1,239 @@
+{
+    Access = Internal;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        RenciSFTPClient: DotNet RenciSftpClient;
+        LastOperationSuccessful: Boolean;
+
+    procedure Disconnect()
+    begin
+        RenciSFTPClient.Disconnect();
+    end;
+
+    procedure IsConnected(): Boolean
+    begin
+        exit(RenciSFTPClient.IsConnected());
+    end;
+
+    procedure Exists(Path: Text; var ExistResult: Boolean): Boolean
+    begin
+        LastOperationSuccessful := InternalExists(Path, ExistResult);
+        exit(LastOperationSuccessful);
+    end;
+
+    [TryFunction]
+    local procedure InternalExists(Path: Text; var ExistResult: Boolean)
+    begin
+        ExistResult := RenciSFTPClient.Exists(Path);
+    end;
+
+    procedure Delete(Path: Text): Boolean
+    begin
+        LastOperationSuccessful := InternalDelete(Path);
+        exit(LastOperationSuccessful);
+    end;
+
+    [TryFunction]
+    local procedure InternalDelete(Path: Text)
+    begin
+        RenciSFTPClient.Delete(Path);
+    end;
+
+    procedure WorkingDirectory(var Result: Text): Boolean
+    begin
+        LastOperationSuccessful := InternalWorkingDirectory(Result);
+        exit(LastOperationSuccessful);
+    end;
+
+    [TryFunction]
+    local procedure InternalWorkingDirectory(var Result: Text)
+    begin
+        Result := RenciSFTPClient.WorkingDirectory;
+    end;
+
+    [NonDebuggable]
+    procedure SftpClient(Host: Text; Port: Integer; UserName: Text; Password: SecretText): Boolean
+    begin
+        RenciSFTPClient := RenciSftpClient.SftpClient(Host, Port, UserName, Password.Unwrap());
+        LastOperationSuccessful := InternalConnect();
+        exit(LastOperationSuccessful);
+    end;
+
+    procedure SftpClient(Host: Text; Port: Integer; UserName: Text; PrivateKey: InStream): Boolean
+    var
+        PrivateKeyFile: DotNet RenciPrivateKeyFile;
+        Arr: DotNet Array;
+    begin
+        PrivateKeyFile := PrivateKeyFile.PrivateKeyFile(PrivateKey);
+        Arr := Arr.CreateInstance(GetDotNetType(PrivateKeyFile), 1);
+        Arr.SetValue(PrivateKeyFile, 0);
+        RenciSftpClient := RenciSftpClient.SftpClient(Host, Port, Username, Arr);
+        LastOperationSuccessful := InternalConnect();
+        exit(LastOperationSuccessful);
+    end;
+
+    [NonDebuggable]
+    procedure SftpClient(HostName: Text; Port: Integer; Username: Text; PrivateKey: InStream; Passphrase: SecretText): Boolean
+    var
+        PrivateKeyFile: DotNet RenciPrivateKeyFile;
+        Arr: DotNet Array;
+    begin
+        PrivateKeyFile := PrivateKeyFile.PrivateKeyFile(PrivateKey, Passphrase.Unwrap());
+        Arr := Arr.CreateInstance(GetDotNetType(PrivateKeyFile), 1);
+        Arr.SetValue(PrivateKeyFile, 0);
+        RenciSftpClient := RenciSftpClient.SftpClient(HostName, Port, Username, Arr);
+        LastOperationSuccessful := InternalConnect();
+        exit(LastOperationSuccessful);
+    end;
```

Comment:
I would have the first `SftpClient` method call the second with an empty secrettext for passphrase. These 2 are exactly the same aside from the passphrase addition.

I'm not familiar with RenciPrivateKeyFile whether an empty passphrase is the same as no passphrase.

If so, then there should be no problems. Otherwise we can always have an if/else check `Passphrase.IsEmpty()`.

Reply:
I will try if that is the case!